src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+package io.mosip.print.service.impl;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.print.constant.ApiName;
+import io.mosip.print.constant.LoggerFileConstant;
+import io.mosip.print.exception.ApisResourceAccessException;
+import io.mosip.print.exception.PlatformErrorMessages;
+import io.mosip.print.logger.PrintLogger;
+import io.mosip.print.service.PrintRestClientService;
+import io.mosip.print.util.RestApiClient;
+
+/**
+ * The Class RegistrationProcessorRestClientServiceImpl.
+ * 
+ * @author Rishabh Keshari
+ */
+@Service
+public class PrintRestClientServiceImpl implements PrintRestClientService<Object> {
+
+	/** The logger. */
+	private static Logger regProcLogger = PrintLogger
+			.getLogger(PrintRestClientServiceImpl.class);
+
+	/** The rest api client. */
+	@Autowired
+	private RestApiClient restApiClient;
+
+	/** The env. */
+	@Autowired
+	private Environment env;
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see io.mosip.registration.processor.core.spi.restclient.
+	 * RegistrationProcessorRestClientService#getApi(io.mosip.registration.
+	 * processor .core.code.ApiName,
+	 * io.mosip.registration.processor.core.code.RestUriConstant, java.lang.String,
+	 * java.lang.String, java.lang.Class)
+	 */
+	@Override
+	public Object getApi(ApiName apiName, List<String> pathsegments, String queryParamName, String queryParamValue,
+			Class<?> responseType) throws ApisResourceAccessException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"RegistrationProcessorRestClientServiceImpl::getApi()::entry");
+		Object obj = null;
+		String apiHostIpPort = env.getProperty(apiName.name());
```

Comment:
I think we should get environment variable with a static method from an utility class or general config to avoid inconsistencies config spread across the source code.